### DISCUSSION TITLE
Does next.js need its own server?

### DISCUSSION THREAD

**Discussion by bexoss** [vercel/next.js#13424] [Help] [answered]
The app based on `create-react-app`, I can build client code and without its server, I can run this app with just one express server (express will serve /build/.html). 

In next.js, I read some official documents, It's seem to I should run next.js server. (npm run build => npm run start). 

So, should I run at least 2 node.js instance for running next.js based app?

**Comment by Timer** [comment, marked as answer]
You should host them separately, yes.

However, take a look at Next.js' API routes feature:
https://nextjs.org/docs/api-routes/introduction
